cli/commands/auth/login.ts

```diff
@@ -0,0 +1,96 @@
+import { password } from '@inquirer/prompts';
+import { __, sprintf } from '@wordpress/i18n';
+import { getAuthenticationUrl } from 'common/lib/oauth';
+import { AuthCommandLoggerAction as LoggerAction } from 'common/logger-actions';
+import { getUserInfo } from 'cli/lib/api';
+import {
+	getAuthToken,
+	lockAppdata,
+	readAppdata,
+	saveAppdata,
+	unlockAppdata,
+} from 'cli/lib/appdata';
+import { openBrowser } from 'cli/lib/browser';
+import { getAppLocale } from 'cli/lib/i18n';
+import { Logger, LoggerError } from 'cli/logger';
+import { StudioArgv } from 'cli/types';
+
+const CLI_REDIRECT_URI = `https://developer.wordpress.com/copy-oauth-token`;
+
+export async function runCommand(): Promise< void > {
+	const logger = new Logger< LoggerAction >();
+
+	try {
+		await getAuthToken();
+		logger.reportSuccess( __( 'Already authenticated with WordPress.com' ) );
+		return;
+	} catch ( error ) {
+		// Assume the token is invalid and proceed with authentication
+	}
+
+	logger.reportStart( LoggerAction.LOGIN, __( 'Opening browser for authentication…' ) );
+
+	const appLocale = await getAppLocale();
+	const authUrl = getAuthenticationUrl( appLocale, CLI_REDIRECT_URI );
+
+	try {
+		await openBrowser( authUrl );
+		logger.reportSuccess( __( 'Browser opened successfully' ) );
+	} catch ( error ) {
+		// If the browser fails to open, allow users to manually open the URL
+		const loggerError = new LoggerError(
+			sprintf( __( 'Failed to open browser. Please open the URL manually: %s' ), authUrl ),
+			error
+		);
+		logger.reportError( loggerError );
+	}
+
+	console.log(
+		__( 'Please complete authentication in your browser and paste the generated token here.' )
+	);
+	console.log( '' );
+
+	try {
+		const accessToken = await password( { message: __( 'Authentication token:' ) } );
+		const user = await getUserInfo( accessToken );
+
+		logger.reportSuccess( __( 'Authentication completed successfully!' ) );
+
+		try {
+			await lockAppdata();
+			const userData = await readAppdata();
+
+			const now = new Date();
+			const twoWeeksInSeconds = 2 * 7 * 24 * 60 * 60;
```

Comment:
I just saw that the `expiresIn` in appData is `1209600` seconds or two weeks, but I always thought the token was valid for months. 🤔